rust/ql/lib/codeql/rust/frameworks/Sqlx.qll

```diff
@@ -0,0 +1,40 @@
+/**
+ * Provides modeling for the `SQLx` library.
+ */
+
+private import rust
+private import codeql.rust.Concepts
+private import codeql.rust.dataflow.DataFlow
+
+/**
+ * A call to `sqlx::query` and variations.
+ */
+private class SqlxQuery extends SqlConstruction::Range {
+  CallExpr call;
+
+  SqlxQuery() {
+    this.asExpr().getExpr() = call and
+    call.getFunction().(PathExpr).getPath().getResolvedPath() =
+      [
+        "crate::query::query", "crate::query_as::query_as", "crate::query_with::query_with",
+        "crate::query_as_with::query_as_with", "crate::query_scalar::query_scalar",
+        "crate::query_scalar_with::query_scalar_with", "crate::raw_sql::raw_sql"
+      ]
+  }
+
+  override DataFlow::Node getSql() { result.asExpr().getExpr() = call.getArgList().getArg(0) }
+}
+
+/**
+ * A call to `sqlx::Executor::execute`.
+ */
+private class SqlxExecute extends SqlConstruction::Range {
```

Comment:
Based on the name alone I would've thought that this would implement `SqlExecution::Range`?

Reply:
Yes I think you're right.  There are actually three possible points of interest:
1. where an SQL string is compiled into some kind of object
2. where an SQL string is compiled and executed immediately
3. where a previously compiled object is executed

I had thought `SqlConstruction` was supposed to be the first two and `SqlExecution` was the third one - but looking at the API (which is the same as other languages have) `SqlExecution` is clearly supposed to be the second one, `SqlConstruction` just the first, and we have nothing for the third (but perhaps not that interesting for analysis).

I'll update the code...

Reply:
Fixed.